pywr/solvers/cython_glpk.pyx

```diff
@@ -186,28 +250,43 @@ cdef class CythonGLPKSolver:
         cdef int status
         cdef cross_domain_col
         cdef list route
+        cdef int node_id
+        cdef double flow
 
-        t0 = time.clock()
         timestep = model.timestep
         routes = self.routes
         non_storages = self.non_storages
         storages = self.storages
 
-        # update route properties
+        # update route cost
+
+        t0 = time.clock()
+
+        # update the cost of each node in the model
+        cdef double[:] node_costs = self.node_costs_arr
+        for _node in self.nodes_with_cost:
+            data = _node.__data
```

Comment:
I wanted to keep this agnostic to the solver. Try changing it and see if it makes a difference?